### ISSUE TITLE
Calculate Gain Importance on Test Data

### ISSUE THREAD

**Issue by brunogasperini** [microsoft/LightGBM#2868]
## Summary

I want to calculate the Gain feature importance from a LightGBM model on a test dataset.

I did research and was not able to do it using LightGBM - it can only calculate these feature importance from the model (which was obviously estimated in the training sample). 

I know that XGBoost has somewhat recently released a method that does allow to do.

Do you plan to have this feature for LightGBM as well?

Thank you!

**Comment by StrikerRUS** [comment]
@brunogasperini Hey!

> I know that XGBoost has somewhat recently released a method that does allow to do.

Do you have any references?

Is this the same as #2413?

**Comment by brunogasperini** [comment]
@StrikerRUS 

Yes - XGBoost has an ability to 'refresh' the node statistics (gain, cover, and value) using any dataset. So it recompute the gains using the test set.

https://github.com/dmlc/xgboost/pull/1670

**Comment by StrikerRUS** [comment]
Thank you @brunogasperini for the clarification! Then I'm closing this issue as a duplicated one.
